config/application.rb

```diff
@@ -45,9 +45,6 @@ class Application < Rails::Application
         load override
       end
     end
-
-    # Load rake tasks
-    Rails.application.load_tasks
```

Comment:
This is the culprit. In order for us to call rake tasks within the system we need to load the available rake tasks.

This means when we load/start bundle/server, we load all rake tasks. Running the task through bundle exec rake manually is first loading the rails application.rb (which loads the rake tasks) and then loads the task we passed to it, so it is executed twice. Removing this line should fix the issue.

Reply:
Any idea why each task was running twice on invocation?  I would assume that `load_tasks` only declares the tasks; does the rake task tracking system do something more complicated than associating task names with the blocks that execute them within its dependency graph?

Reply:
There are several github discussions on that in the past (maybe it was already patched in newer version of rails), but when we load rake task from application.rb and thru bundle exec rake, it calls all matching rake task that was loaded (in our case there will be two of them)